dashboard/tests/test_model_layer.py

```diff
@@ -25,4 +35,162 @@ def test_get_sessions_country_stats(self):
+        self._create_session(
+            'session key 1', 'node key', 'service type', now,  1, 2
+        )
+        self._create_session(
+            'session key 2', 'node key', 'service type', now, 3, 4
+        )
+        self._create_session(
+            'session key 3', 'node key', 'other service type', now, -100, -100
+        )
+        self._create_session(
+            'session key 4', 'other node key', 'service type', now, -100, -100
+        )
+        _, total_bytes = aggregate_node_data_from_sessions(
+            'node key',
+            'service type',
+            now,
+            now + second
+        )
+        self.assertEqual(10, total_bytes)
+
+    def test_where_dates_mismatch(self):
+        self._create_session(
+            'session key 1', 'node key', 'service type', now - second, 1, 2
+        )
+        self._create_session(
+            'session key 2', 'node key', 'service type', now + second, 1, 2
+        )
+        _, total_bytes = aggregate_node_data_from_sessions(
+            'node key',
+            'service type',
+            now,
+            now + second
+        )
+        self.assertEqual(0, total_bytes)
+
+    @staticmethod
+    def _create_session(session_key, node_key, service_type,
+                        updated_at, bytes_sent, bytes_received):
+        session = models.Session(session_key)
+        session.node_key = node_key
+        session.service_type = service_type
+        session.client_updated_at = updated_at
+        session.client_bytes_sent = bytes_sent
+        session.client_bytes_received = bytes_received
+        get_db().session.add(session)
+
+
+class TestAggregateSessionsCount(TestCase):
+    def test_match_query_params(self):
+        self._create_session(
+            'session key 1', 'node key', 'service type', now
+        )
+        self._create_session(
+            'session key 2', 'other node key', 'service type', now
+        )
+        self._create_session(
+            'session key 3', 'node key', 'other service type', now
+        )
+        self._create_session(
+            'session key 4', 'node key', 'service type', now - second
+        )
+        self._create_session(
+            'session key 5', 'node key', 'service type', now + second
+        )
+        self._create_session(
+            'session key 6', 'node key', 'service type', now
+        )
+        sessions_count, _ = aggregate_node_data_from_sessions(
+            'node key',
+            'service type',
+            now,
+            now + second
+        )
+        self.assertEqual(2, sessions_count)
+
+    @staticmethod
+    def _create_session(session_key, node_key, service_type, updated_at):
+        session = models.Session(session_key)
+        session.node_key = node_key
+        session.service_type = service_type
+        session.client_updated_at = updated_at
+        get_db().session.add(session)
+
+
+class TestGetNodeHoursOnline(TestCase):
+    def test_returns_one_hour(self):
```

Comment:
I though `31` had something to do with days in the months, but actually `get_node_hours_online` returns rounded result - we might specify this behaviour we're testing in method name.
```suggestion
    def test_result_is_rounded_to_hour(self):
```

Reply:
`+`